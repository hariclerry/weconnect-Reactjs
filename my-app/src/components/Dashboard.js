import React from 'react';
import './static/css/bootstrap.min.css'


export default class Dashboard extends React.Component {
    render () {
        return (
            <div>
                <nav className="navbar navbar-inverse">
                <div className="container-fluid">
                  <div className="navbar-header">
                    <a className="navbar-brand" href="/">WeConnect</a>
                  </div>
                 
                  
                    <ul className="nav navbar-nav navbar-right">
                        <li><a href=""><span className="glyphicon glyphicon-user"></span>User</a></li>
                        <li><a href="/"><span className="glyphicon glyphicon-log-out"></span> Logout</a></li>
                          </ul>
        <ul className="nav navbar-nav">
            <li><a href="/">Home</a></li>
            <li>
                <a href="/dashboard">Dasboard</a>
            </li>
            <li>
                <a href="/registerbusiness">Register Business</a>
            </li>
            <li>
                <a href="/businesslist">All Businesses</a>
            </li>
            <li>
                <a href="/singlebusiness">Single Business</a>
            </li>
                    </ul>
              </div>
              </nav>
    
                    <div className="container">
                       
              <div className="table-responsive">
                      <table className="table table-striped">
                        
                          <h4>Hello User, welcome to your dashboard, Below are a list of your Registered businesses</h4>
                         <hr/>
                          <thead> 
                                  <td><b>Name</b></td>
                                  <td><b>Category</b></td>
                                  <td><b>Location</b></td>
                                  <td><b>Description</b></td>
                                  <td><b>Reviews</b></td>
                                  <td><b>Options</b></td>
                                  
                          </thead>
                          <tbody >
                             
                              <tr>
                                  <td>Andela</td>
                                  <td>Technology</td>
                                  <td>Kampala</td>
                                  <td>This is Andela</td>
                                  <td>
                                  
                                      <a href="review.html" className="btn btn-info" role="button">View Reviews</a>
                                  </td>
                                  <td>
                                      <a href="updatebusiness" className="btn btn-info" role="button" style={{float: "left"}}>Edit</a>
                                      <a href="delete" className="btn btn-info" role="button" style={{float: "right"}}>Delete</a>
                              
                                  </td>
                                 
                              </tr>
                             
                          </tbody>
                         
                       </table>
                  </div>
                  </div>

    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <footer className="row">
           
            <div className="panel-footer text-center bg-dark">
        <p>  &copy; Copyright 2018 <i>powered by</i>  <b> InfoClan</b> </p>
            </div>
    
    </footer>
            </div>

        )
    }
}