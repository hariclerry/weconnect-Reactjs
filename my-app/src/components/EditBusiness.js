import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

// import 'react-notifications/lib/notifications.css';

import { NotificationManager } from 'react-notifications';
import './static/css/bootstrap.min.css'
import { editBusiness} from '../actions/businessActions';
import {checkIfUserIsLoggedIn} from '../actions/userActions';
import './static/css/bootstrap.min.css'
import NavBar from './NavBar';
import Footer from './Footer';
// TO DO: import {checkIfUserIsLoggedIn} from '../actions/userActions';


class EditBusiness extends React.Component {

  componentWillReceiveProps(recieved) {
    if (recieved && recieved.business.message === "Business successfully registered") {
        this.props.history.push('/dashboard');
    }
    else{
        if(recieved && recieved.business.status === "failure"){
            NotificationManager.error(recieved.business.message, "", 5000);
        }
    }
};


componenWillMount() {
    checkIfUserIsLoggedIn(this.props.email,this.props.history);

}

jsonStringify = object =>{
    let simpleObj={};
        for (let prop in object){
                if (!object.hasOwnProperty(prop)){
                        continue;
                    }
                if (typeof(object[prop]) === 'object'){
                        continue;
                    }
                simpleObj[prop] = object[prop];
            }
            return JSON.stringify(simpleObj)}

updateBusiness = e => {
    e.preventDefault(); // Prevent link from opening the URL(synthetic event):
    let businessCredential = {
        name: e.target.elements.name.value,
        category: e.target.elements.category.value,
        location: e.target.elements.location.value,
        description: e.target.elements.description.value
    };
  
    // this.props.editBusiness(this.jsonStringify(businessCredential))

}
    render () {
        console.log(this.props.business)
        console.log(this.props.email)
        return (
    <div>
            <NavBar />
    
             
                  

                        <div className="container">
                               <h3> You can now update your Business</h3>
                        </div>
                        <hr/>
                        <div style={{margin: "100px"}}>
                        <form onSubmit={this.updateBusiness }>
                          
                            <div className="form-group"> 
                              <label className="control-label requiredField" for="title">Business Name:</label>
                              <input className="form-control" id="title" name="title" type="text"/>
                            </div>

                            <div className="form-group">
                                    <label className="control-label " for="category">Business Category:</label>
                                    <input className="form-control" id="category" name="category" type="text"/>
                            </div>
                            
                            <div className="form-group">
                              <label className="control-label " for="location">Business Location:</label>
                              <input className="form-control" id="location" name="location" type="text"/>
                            </div>
                            
                            
                            <div className="form-group"> 
                              <label className="control-label " for="description">Business Description:</label>
                              <textarea className="form-control" cols="40" id="description" name="description" rows="10"></textarea>
                            </div>
                            
                            <div className="form-group">
                              <button className="btn btn-primary " name="submit" type="submit">Update</button>
                            </div>
                            
                          </form>	
                          </div>						
                                      
                       
                        
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <Footer />
     </div>

        )
    }
}

EditBusiness.protoTypes = {
  email: PropTypes.string.isRequired,
  editBusiness: PropTypes.func.isRequired
}

const mapStateToProps = state =>{
console.log(state);
return {
  email: state.auth.loginData.email,
  business: state.business.singleBusinessMessage.business_data,

}};

export default  withRouter(connect(mapStateToProps,{editBusiness })(EditBusiness));