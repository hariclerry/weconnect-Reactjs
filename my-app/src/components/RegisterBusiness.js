import React from 'react';

import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

// import 'react-notifications/lib/notifications.css';

import { NotificationManager } from 'react-notifications';
import './static/css/bootstrap.min.css'
import { registerBusiness} from '../actions/businessActions';
import Footer from './Footer';


class RegisterBusiness extends React.Component {

    componentWillReceiveProps(recieved) {
        if (recieved && recieved.business.message === "Business successfully registered") {
            this.props.history.push('/dashboard');
        }
        else{
            if(recieved && recieved.business.status === "failure"){
                NotificationManager.error(recieved.business.message, "", 5000);
            }
        }
};

	jsonStringify = object =>{
        let simpleObj={};
            for (let prop in object){
                    if (!object.hasOwnProperty(prop)){
                            continue;
                        }
                    if (typeof(object[prop]) === 'object'){
                            continue;
                        }
                    simpleObj[prop] = object[prop];
                }
                return JSON.stringify(simpleObj)}

    createBusiness = e => {
        e.preventDefault(); // prevent the default link behavior of opening a new page
        let businessCredential = {
            name: e.target.elements.name.value,
            category: e.target.elements.category.value,
            location: e.target.elements.location.value,
            description: e.target.elements.description.value
        };
        this.props.registerBusiness(this.jsonStringify(businessCredential))

    }
    render () {
        return (
            <div>
                 <nav className="navbar navbar-inverse">
    <div className="container-fluid">
      <div className="navbar-header">
        <a className="navbar-brand" href="/">WeConnect</a>
      </div>
     
      
        <ul className="nav navbar-nav navbar-right">
            <li><a href=""><span className="glyphicon glyphicon-user"></span>User</a></li>
            <li><a href="index.html"><span className="glyphicon glyphicon-log-out"></span> Logout</a></li>
              </ul>
       <ul className="nav navbar-nav">
       <li><a href="/">Home</a></li>
            <li>
                <a href="/dashboard">Dasboard</a>
            </li>
            <li>
                <a href="/registerbusiness">Register Business</a>
            </li>
            <li>
                <a href="/businesslist">All Businesses</a>
            </li>
            <li>
                <a href="/singlebusiness">Single Business</a>
            </li>
        </ul>
  </div>
  </nav>
    
             
                  

                        <div className="container">
                               <h3> You can now Register your Business</h3>
                        </div>
                        <hr/>
                        <div style={{margin: "100px"}}>
                        <form onSubmit={this.createBusiness }>
                          
                            <div className="form-group"> 
                              <label className="control-label requiredField" htmlFor="title">Business Name:</label>
                              <input className="form-control"  name="name" id="title" type="text"/>
                            </div>

                            <div className="form-group">
                                    <label className="control-label " htmlFor="category">Business Category:</label>
                                    <input className="form-control"  id="category" name="category" type="text"/>
                            </div>
                            
                            <div className="form-group">
                              <label className="control-label " htmlFor="location">Business Location:</label>
                              <input className="form-control"  id="location" name="location" type="text"/>
                            </div>
                            
                            
                            <div className="form-group"> 
                              <label className="control-label " htmlFor="description">Business Description:</label>
                              <textarea className="form-control" cols="40" id="description" name="description" rows="10"></textarea>
                            </div>
                            
                            <div className="form-group">
                              <button className="btn btn-primary " name="submit" type="submit">Create</button>
                            </div>
                            
                          </form>	
                          </div>						
                                      
                       
                        
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <Footer />

    </div>

        )
    }
}


RegisterBusiness.protoTypes = {
    registerBusiness: PropTypes.func.isRequired
}

const mapStateToProps = state =>({
    business: state.business.createBusinessMessage

});

export default  withRouter(connect(mapStateToProps,{ registerBusiness })(RegisterBusiness));