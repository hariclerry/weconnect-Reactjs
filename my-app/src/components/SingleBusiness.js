import React from 'react';
import './static/css/bootstrap.min.css'

import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import Loader from 'react-loader';
// import { NavLink } from 'react-router-dom';

// TO DO:import { NotificationManager } from "react-notifications";
import { singleBusiness, deleteBusiness } from "../actions/businessActions";
import {checkIfUserIsLoggedIn} from '../actions/userActions';
import NavBar from "./NavBar";
import Footer from './Footer';


class SingleBusiness extends React.Component {

       state = {
         loaded: false
       }
      componentWillMount() {
        
        checkIfUserIsLoggedIn(this.props.email,this.props.history);
        const { id } = this.props.match.params;
        console.log(id);
        this.props.singleBusiness(id).then(()=>{});
        this.setState({loaded: true})
      }

      deleteBusiness = id  =>{
        if (window.confirm('Are you sure you want to delete this business?')) {
                this.props.deleteBusiness(id).then(()=>{
                  console.log(" confirmed deleted........")
                });
               this.props.history.push("/dashboard");
        }
    }
    render () {
        // Deconstructing an object
        const{business} = this.props;
        const {loaded} = this.state;

        return (
    <div>
    <NavBar />
            <div className="container">
            <h4>Hello User</h4>
            <hr/>

        <div className="col-md-8 col-md-offset-2">
        <Loader loaded={loaded}>
        <h1>{business.name}</h1>
        <p><b>Category:   </b> {business.category}</p>
        <p><b>Location:   </b>{business.location}</p>
        <p><b>Description:  </b>{business.description}</p>

        <a href="/addreview" className="btn btn-info" role="button" style={{margin: "2%"}}>Review</a>
        <a href="/edit" className="btn btn-info" role="button" color="green" style={{margin: "2%"}}>Edit</a>
        <button onClick={() => this.deleteBusiness(business.id) } className="btn btn-default"  role="button" style={{margin: "2%"}}> Delete </button>
        </Loader>
      </div>

     </div>

    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <Footer />
     </div>

        )
    }
}
SingleBusiness.protoTypes = {
    email: PropTypes.string.isRequired,
    singleBusiness: PropTypes.func.isRequired,
    deleteBusiness: PropTypes.func.isRequired
  };

function mapStateToProps(state, ownProps){
  console.log(state.business.singleBusinessMessage.business_data);
   return {
    email: state.auth.loginData.email,
    business: state.business.singleBusinessMessage.business_data,
    businessId: ownProps.match.params.id,
    deletedBusiness: state.business.deleteBusinessMessage,

   }

  };

  // const mapDispatchToProps = dispatch => ({

  //     loadBusiness: (businessId) => dispatch(singleBusiness(businessId))
  // });

  export default withRouter(connect(mapStateToProps, {singleBusiness, deleteBusiness})(SingleBusiness));