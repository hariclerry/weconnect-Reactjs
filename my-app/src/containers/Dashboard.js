import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";

import { checkIfUserIsLoggedIn } from "../actions/userActions";
import { userBusinesses } from "../actions/businessActions";
import Dashboard from "../components/dashboard";

// TO DO: import { NotificationManager } from "react-notifications";

class UserDashboard extends React.Component {
  componentDidMount() {
    this.props.userBusinesses(localStorage.getItem("user_id"));
    checkIfUserIsLoggedIn(this.props.email, this.props.history);
  }

  render() {
    // Display all businesses from props into a dict
    const businesses = Object.values({ ...this.props.business.business_data });

    // sorts/arranges business from recently added to previous
    if (businesses) {
      Array.prototype.reverse.call(businesses);
    }

    return (
      <div>
      <Dashboard 
      businesses={businesses}
      username={this.props.username}
      />
      </div>
    );
  }
}

// Validate props

Dashboard.propTypes = {
  email: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  userBusinesses: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  email: state.auth.loginData.email,
  username: state.auth.loginData.username,
  business: state.business.userBusinessMessage
});

export default connect(
  mapStateToProps,
  { checkIfUserIsLoggedIn, userBusinesses }
)(UserDashboard);
