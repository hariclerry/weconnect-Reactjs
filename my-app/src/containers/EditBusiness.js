import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";

// import 'react-notifications/lib/notifications.css';

import { NotificationManager } from "react-notifications";
import EditBusiness from "../components/EditBusiness";
import { jsonStringify } from '../utils/jsonHelper'
import "../components/static/css/bootstrap.min.css";
import { editBusiness } from "../actions/businessActions";
import { checkIfUserIsLoggedIn } from "../actions/userActions";
import "../components/static/css/bootstrap.min.css";
// TO DO: import {checkIfUserIsLoggedIn} from '../actions/userActions';

class EditBusinessContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      name: "",
      category: "",
      location: "",
      description: ""
    };
  }

  handleChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };
  componenWillMount() {
    checkIfUserIsLoggedIn(this.props.email, this.props.history);
  }

  updateBusiness = e => {
    e.preventDefault(); // Prevent link from opening the URL(synthetic event):
    let businessCredential = {
      name: e.target.elements.name.value,
      category: e.target.elements.category.value,
      location: e.target.elements.location.value,
      description: e.target.elements.description.value
    };
    const businessId = this.props.match.params.id;
    this.props.editBusiness(businessId, jsonStringify(businessCredential));
  };
  render() {
    console.log(this.state);
    return (
      <EditBusiness
        {...this.props.business}
        handleChange={this.handleChange}
        updateBusiness={this.updateBusiness}
      />
    );
  }
}

EditBusiness.protoTypes = {
  email: PropTypes.string.isRequired,
  editBusiness: PropTypes.func.isRequired
};

const mapStateToProps = state => {
  return {
    email: state.auth.loginData.email,
    business: state.business.singleBusinessMessage.business_data
  };
};

export default withRouter(
  connect(
    mapStateToProps,
    { editBusiness }
  )(EditBusinessContainer)
);
