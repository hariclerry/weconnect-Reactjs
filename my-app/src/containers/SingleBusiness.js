import React from "react";

import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
// import { NavLink } from 'react-router-dom';

// TO DO:import { NotificationManager } from "react-notifications";
import SingleBusiness from "../components/SingleBusiness";
import { singleBusiness, deleteBusiness } from "../actions/businessActions";
import { checkIfUserIsLoggedIn } from "../actions/userActions";

class Business extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      loaded: false
    };
  }

  componentDidMount() {
    checkIfUserIsLoggedIn(this.props.email, this.props.history);
    const { id } = this.props.match.params;
    this.props.singleBusiness(id);
    // this.setState({loaded: true})
  }

  deleteBusiness = id => {
    if (window.confirm("Are you sure you want to delete this business?")) {
      this.props.deleteBusiness(id).then(() => {
        console.log(" confirmed deleted........");
      });
      this.props.history.push("/dashboard");
    }
  };
  render() {
    const { business } = this.props;
    const { loaded } = this.state;
    return (
      <SingleBusiness
        business={business}
        deleteBusiness={this.deleteBusiness}
        loaded={loaded}
      />
    );
  }
}
SingleBusiness.protoTypes = {
  email: PropTypes.string.isRequired,
  singleBusiness: PropTypes.func.isRequired,
  deleteBusiness: PropTypes.func.isRequired
};

const mapStateToProps = (state, ownProps) => {
  console.log(state);
  return {
    email: state.auth.loginData.email,
    business: state.business.singleBusinessMessage.business_data,
    businessId: ownProps.match.params.id,
    deletedBusiness: state.business.deleteBusinessMessage
  };
};

const mapDispatchToProps = dispatch => ({
  singleBusiness: id => dispatch(singleBusiness(id)),
  deleteBusiness: id => dispatch(deleteBusiness(id))
});

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(Business)
);
