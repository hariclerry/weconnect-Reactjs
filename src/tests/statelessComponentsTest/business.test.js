import React from "react";
import {shallow} from "enzyme";
import RegisterBusiness from "components/business/registerBusiness";
// import Businesses from "components/business/businessList";
import ReviewBusiness from "components/business/addReview";
import Dashboard from "components/business/dashboard";
import EditBusiness from "components/business/editBusiness";
import Pagination from "components/business/pagination";
import SearchBusiness from "components/business/searchBar";
import SingleBusiness from "components/business/singleBusiness";



describe('RegisterBusiness component',() => {

  it('should match snapshot',() =>{
      const wrapper = shallow(<RegisterBusiness />)
      expect(wrapper).toMatchSnapshot();
  })
})

// describe('Businesses component',() => {
//   const props = {
//     map: []
//     }

//   it('should match snapshot',() =>{
//       const wrapper = shallow(<Businesses {...props}/>)
//       expect(wrapper).toMatchSnapshot();
//   })
// })


describe('ReviewBusiness component',() => {
  const props = {
    businessName:[]
    }

  it('should match snapshot',() =>{
      const wrapper = shallow(<ReviewBusiness {...props}/>)
      expect(wrapper).toMatchSnapshot();
  })
})

describe('Dashboard component',() => {

  it('should match snapshot',() =>{
      const wrapper = shallow(<Dashboard />)
      expect(wrapper).toMatchSnapshot();
  })
})

describe('EditBusiness component',() => {

  it('should match snapshot',() =>{
      const wrapper = shallow(<EditBusiness />)
      expect(wrapper).toMatchSnapshot();
  })
})

describe('Pagination component',() => {

  it('should match snapshot',() =>{
      const wrapper = shallow(<Pagination />)
      expect(wrapper).toMatchSnapshot();
  })
})

describe('SearchBusiness component',() => {

  it('should match snapshot',() =>{
      const wrapper = shallow(<SearchBusiness />)
      expect(wrapper).toMatchSnapshot();
  })
})

describe('SingleBusiness component',() => {
  const props = {
    business:[]
    }

  it('should match snapshot',() =>{
      const wrapper = shallow(<SingleBusiness {...props}/>)
      expect(wrapper).toMatchSnapshot();
  })
})

