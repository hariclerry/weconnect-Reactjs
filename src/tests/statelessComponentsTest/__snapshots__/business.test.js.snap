// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component should match snapshot 1`] = `
<div>
  <_class
    options={Object {}}
  />
  <withRouter(Connect(NavBar)) />
  <div
    className="container"
  >
    <div
      style={
        Object {
          "marginBottom": "150px",
        }
      }
    >
      <h4
        style={
          Object {
            "float": "left",
          }
        }
      >
        Hello 
        , welcome to your dashboard
      </h4>
      <Link
        className="btn btn-success"
        replace={false}
        role="button"
        style={
          Object {
            "float": "right",
            "margin": "0",
          }
        }
        to="/resetpassword/"
      >
        Reset Password
      </Link>
    </div>
    <hr />
    <br />
    <div
      className="fixed-footer"
    >
      <div
        className="table-responsive"
      >
        <table
          className="table table-striped"
        >
          <thead>
            <td>
              <b>
                Name
              </b>
            </td>
            <td>
              <b>
                Category
              </b>
            </td>
            <td>
              <b>
                Location
              </b>
            </td>
            <td>
              <b>
                Description
              </b>
            </td>
            <td>
              <b>
                Review
              </b>
            </td>
          </thead>
          <tbody>
            <div>
               No Businesses found 
            </div>
          </tbody>
        </table>
      </div>
    </div>
  </div>
  <Footer />
</div>
`;

exports[`EditBusiness component should match snapshot 1`] = `
<div>
  <withRouter(Connect(NavBar)) />
  <div
    className="container"
  >
    <h3>
       You can now update your Business
    </h3>
  </div>
  <hr />
  <div
    style={
      Object {
        "margin": "100px",
      }
    }
  >
    <form>
      <div
        className="form-group"
      >
        <label
          className="control-label requiredField"
          for="title"
        >
          Business Name:
        </label>
        <input
          className="form-control"
          id="title"
          name="name"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="control-label "
          for="category"
        >
          Business Category:
        </label>
        <input
          className="form-control"
          id="category"
          name="category"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="control-label "
          for="location"
        >
          Business Location:
        </label>
        <input
          className="form-control"
          id="location"
          name="location"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="control-label "
          for="description"
        >
          Business Description:
        </label>
        <textarea
          className="form-control"
          cols="40"
          id="description"
          name="description"
          required={true}
          rows="10"
        />
      </div>
      <div
        className="form-group"
      >
        <button
          className="btn btn-primary "
          name="submit"
          type="submit"
        >
          Update
        </button>
      </div>
    </form>
  </div>
  <Footer />
</div>
`;

exports[`Pagination component should match snapshot 1`] = `
<nav
  aria-label="Page navigation example"
  style={
    Object {
      "margin": "0 500px",
    }
  }
>
  <ul
    className="pagination justify-content-center"
  >
    <li
      className="page-item"
      key="prev"
    >
      <button
        className="page-link"
        href=""
        name="prevPage"
        onClick={[Function]}
      >
        prev
      </button>
    </li>
    <li
      className="page-item"
      key="next"
    >
      <button
        className="page-link"
        href=""
        name="nextPage"
        onClick={[Function]}
      >
        next
      </button>
    </li>
  </ul>
</nav>
`;

exports[`RegisterBusiness component should match snapshot 1`] = `
<div>
  <_class
    options={Object {}}
  />
  <withRouter(Connect(NavBar)) />
  <div
    className="container"
  >
    <h3>
       You can now Register your Business
    </h3>
  </div>
  <hr />
  <div
    style={
      Object {
        "margin": "100px",
      }
    }
  >
    <form>
      <div
        className="form-group"
      >
        <label
          className="control-label requiredField"
          htmlFor="title"
        >
          Business Name:
        </label>
        <input
          className="form-control"
          id="title"
          name="name"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="control-label "
          htmlFor="category"
        >
          Business Category:
        </label>
        <input
          className="form-control"
          id="category"
          name="category"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="control-label "
          htmlFor="location"
        >
          Business Location:
        </label>
        <input
          className="form-control"
          id="location"
          name="location"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          className="control-label "
          htmlFor="description"
        >
          Business Description:
        </label>
        <textarea
          className="form-control"
          cols="40"
          id="description"
          name="description"
          required={true}
          rows="10"
        />
      </div>
      <div
        className="form-group"
      >
        <button
          className="btn btn-primary "
          name="submit"
          type="submit"
        >
          Create
        </button>
      </div>
    </form>
  </div>
  <Footer />
</div>
`;

exports[`ReviewBusiness component should match snapshot 1`] = `
<div>
  <_class
    options={Object {}}
  />
  <withRouter(Connect(NavBar)) />
  <div
    className="container"
  >
    <div
      style={
        Object {
          "margin": "20px",
        }
      }
    >
      <div
        className="box"
      >
        <h1 />
        <p>
          <b>
            Category: 
          </b>
           
        </p>
        <p>
          <b>
            Location: 
          </b>
        </p>
        <p>
          <b>
            Description: 
          </b>
        </p>
      </div>
    </div>
    <hr />
  </div>
  <div
    style={
      Object {
        "margin": "50px 150px",
      }
    }
  >
    <form>
      <div
        className="form-group"
      >
        <label
          className="control-label "
          htmlFor="description"
        >
          Say Something about this Business:
        </label>
        <textarea
          className="form-control"
          cols="2"
          id="description"
          name="description"
          required={true}
          rows="2"
        />
      </div>
      <div
        className="form-group"
      >
        <button
          className="btn btn-primary "
          name="submit"
          type="submit"
        >
          Add Review
        </button>
      </div>
    </form>
  </div>
  <div
    style={
      Object {
        "margin": "50px 150px",
        "width": "auto",
      }
    }
  >
    <h3>
      Reviews
    </h3>
    <div>
       No Reviews found 
    </div>
  </div>
  <Footer />
</div>
`;

exports[`SearchBusiness component should match snapshot 1`] = `
<div>
  <form
    method="POST"
    onSubmit={[Function]}
  >
    <div>
      <div
        className="col-md-8 col-md-offset-2 businesssearch"
      >
        <div
          className="input-group"
        >
          <input
            className="form-control"
            name="name"
            placeholder="Search business"
            type="text"
          />
          <span
            className="input-group-btn"
          >
            <button
              className="btn btn-info"
              id="search"
              type="submit"
            >
              <i
                className="fa fa-search fa-fw"
              />
               Search
            </button>
          </span>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`SingleBusiness component should match snapshot 1`] = `
<div>
  <withRouter(Connect(NavBar)) />
  <div
    className="container"
  >
    <h4>
      Business
    </h4>
    <hr />
    <div
      className="box"
    >
      <h1>
        hari
      </h1>
      <p>
        <b>
          Category: 
        </b>
         
      </p>
      <p>
        <b>
          Location: 
        </b>
      </p>
      <p>
        <b>
          Description: 
        </b>
      </p>
      <Link
        className="btn btn-success"
        replace={false}
        role="button"
        style={
          Object {
            "marginRight": "20px",
          }
        }
        to="/addreview/undefined"
      >
        Review
      </Link>
      <Link
        className="btn btn-info"
        replace={false}
        role="button"
        style={
          Object {
            "marginRight": "20px",
          }
        }
        to="/editbusiness/undefined"
      >
        Edit
      </Link>
      <button
        className="btn btn-danger"
        onClick={[Function]}
        role="button"
        style={
          Object {
            "margin": "1%",
          }
        }
      >
         
        Delete
         
      </button>
    </div>
  </div>
  <Footer />
</div>
`;
